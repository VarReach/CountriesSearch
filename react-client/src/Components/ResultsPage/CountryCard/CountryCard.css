.results__country-card {
  display: inline-block;
  /* background-color: darkgrey; */
}

.results__country-card__content {
  padding: 0 1em;
}

.results__country-card img {
  max-width: 10em;
}

.results__country-card h4 {
  display: inline-block;
  margin: 0;
}

.results__country-card span {
  display: block;
}

.results__country-card ul {
  margin: 0;
  list-style: none;
}