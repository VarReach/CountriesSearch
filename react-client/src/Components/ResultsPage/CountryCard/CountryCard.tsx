import React, { FC } from "react";
import { Country } from "../../../Services/CountriesService";
import "./CountryCard.css";


export interface CountryCardProps {
  country: Country;
}

export const CountryCard: FC<CountryCardProps> = (props) => {
  const { country } = props;
  return (
    <div className="results__country-card">
      <h3>{country.name}</h3>
      <div className="results__country-card__content">
        <img src={country.flag} />
        <div>
          <span><h4>Alpha Code 2:</h4> {country.alpha2Code}</span>
          <span><h4>Alpha Code 3:</h4> {country.alpha3Code}</span>
        </div>
        <div>
          <span><h4>Region:</h4> {country.region}</span>
          <span><h4>Subregion:</h4> {country.subregion}</span>
        </div>
        <div>
          <span><h4>Population:</h4> {Number(country.population).toLocaleString()}</span>
        </div>
        <div>
          <h4>Languages:</h4>
          <ul>
            {country.languages.map((language, i) => {
              return <li key={i}>{language.name}</li>
            })}
          </ul>
        </div>
      </div>
    </div>
  );
}

export default CountryCard;