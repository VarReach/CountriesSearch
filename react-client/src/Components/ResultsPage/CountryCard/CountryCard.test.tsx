import React from "react";
import { render } from "@testing-library/react";
import mockCountries from "../../../../__mocks__/countriesMock";
// Components
import CountryCard, { CountryCardProps } from "./CountryCard";

/**
 * Covers tests for the ResultsPage CountryCard
 */

describe("<CountryCard/>", () => {
  const defaultProps: CountryCardProps = {
    country: mockCountries[0],
  };

  test("renders without crashing", () => {
    const { container } = render(<CountryCard {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });
});