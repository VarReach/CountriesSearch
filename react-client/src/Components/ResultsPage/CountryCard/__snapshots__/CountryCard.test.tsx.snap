// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CountryCard/> renders without crashing 1`] = `
<div>
  <div
    class="results__country-card"
  >
    <h3>
      Test Country
    </h3>
    <div
      class="results__country-card__content"
    >
      <img
        src="testflag.png"
      />
      <div>
        <span>
          <h4>
            Alpha Code 2:
          </h4>
           
          TS
        </span>
        <span>
          <h4>
            Alpha Code 3:
          </h4>
           
          TST
        </span>
      </div>
      <div>
        <span>
          <h4>
            Region:
          </h4>
           
          AMERICAS
        </span>
        <span>
          <h4>
            Subregion:
          </h4>
           
          Test-land
        </span>
      </div>
      <div>
        <span>
          <h4>
            Population:
          </h4>
           
          1
        </span>
      </div>
      <div>
        <h4>
          Languages:
        </h4>
        <ul>
          <li>
            Test Language
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
